CONTRIBUTING.rst

```diff
@@ -124,15 +124,36 @@ See the relevant sections above on how to do this.
 Setting up the environment
 ^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Follow the steps in the :ref:`installation_instructions` > "From Source" section
-to set up your environment, install the required development dependencies, and
-run the tests.
+To contribute, you will first have to run through some steps:
+
+- Set up your environment by forking the repository ((`Github doc on
+  forking and
+  cloning <https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/fork-a-repo>`__)).
+- Create a new virtual environment either with
+  `venv <https://docs.python.org/3/library/venv.html>`__ or
+  `conda <https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html>`__,
+  then activate it.
+- While at the root of your local copy of skrub and within the new env,
+  install the required development dependencies by running
+  ``pip install --editable ".[dev, lint, test, doc]"``.
+- Run all the tests by running ``pytest -s skrub/tests``.
+- Run ``pre-commit install`` to activate some checks that will run every
+  time you do a ``git commit`` (mostly, formatting checks).
```

Comment:
I think one thing that can be confusing with pre-commit is it can modify some files and cancel the commit, and then you need to stage its modification and run `commit` again. I'm not sure what would be the best place to add a little explanation about that without making this section too long :thinking:

Reply:
I added a section on that a bit further down